src/os/README.md

```diff
@@ -104,3 +105,125 @@ inspiration for several decisions in this library.
 - that said, if an operation is extremely efficient and almost always wanted,
   consider doing it natively, automatically. For example, ints in HSL
   `sockaddr` are always in host byte order, not network byte order.
+
+## Appendix:  C / Hack type mapping cheat sheet
+
+| C type | Hack type |
+|---|---|
+| `int` or `short` as a bit set | `keyset<Flag>` where `Flag` is an `enum` |
+| `int socket` or `int filedes`, or other long-lived system resources | `HH\Lib\OS\FileDescriptor` or other wrapper classes |
+| `void *` or `struct *` or other short-lived pointers that do not hold expensive system resource | `shape` or `vec`, see [Appendix: short-lived C pointer encoding](#appendix-short-lived-c-pointer-encoding) |
+
+## Appendix: short-lived C pointer encoding
+
+A common practice in C library is to expose a set of functions, including a
+constructor, a destructor and several setters, to manipulate an short-lived C
+pointer. Instead of directly provide Hack version of these functions, a `shape`
+or `vec` corresponds to the short-lived C pointer should be expose, and the underlying C
+`struct` should be internally created on demand from the `shape` or `vec`, when
+another underlying C function requires the short-lived C pointer.
```

Comment:
I similarly disagree with this; the specific examples below largely make sense, but I don't agree with the implication that all short-lived pointers reasonably fit in one of these two categories

Reply:
Any counter examples?

Reply:
Sorry, I think this is tying into us having different understandings of 'short-lived C pointer', and https://github.com/hhvm/hsl/discussions/169#discussioncomment-1816202 is  the discussion that's needed here